power_grid_model_c/power_grid_model/include/power_grid_model/component/asym_line.hpp

```diff
@@ -0,0 +1,139 @@
+#include "../calculation_parameters.hpp"
+#include "../common/common.hpp"
+#include "../common/three_phase_tensor.hpp"
+
+namespace power_grid_model {
+
+class AsymLine : public Branch {
+  public:
+    using InputType = AsymLineInput;
+    using UpdateType = BranchUpdate;
+    static constexpr char const* name = "asym_line";
+
+    explicit AsymLine(AsymLineInput const& asym_line_input, double system_frequency, double u1, double u2)
+        : Branch{asym_line_input}, i_n_{asym_line_input.i_n}, base_i_{base_power_3p / u1 / sqrt3} {
+        if (cabs(u1 - u2) > numerical_tolerance) {
+            throw ConflictVoltage{id(), from_node(), to_node(), u1, u2};
+        }
+
+        ComplexTensor<asymmetric_t> c_matrix = compute_c_matrix_from_input(asym_line_input);
+        ComplexTensor<asymmetric_t> z_series = compute_z_series_from_input(asym_line_input);
+
+        ComplexTensor<asymmetric_t> empty_tensor;
+
+        if (c_matrix.isZero() || z_series.isZero()) {
+            throw UnsupportedInputDescriptionAsymLine();
+        }
+
+        y_series = inv(z_series);
+        y_shunt = 2 * pi * system_frequency * c_matrix * 1.0i;
+    }
+
+    // override getter
+    double base_i_from() const override { return base_i_; }
+    double base_i_to() const override { return base_i_; }
+    double loading(double /* max_s */, double max_i) const override { return max_i / i_n_; };
+    double phase_shift() const override { return 0.0; }
+    bool is_param_mutable() const override { return false; }
+
+  private:
+    double i_n_;
+    double base_i_;
+    ComplexTensor<asymmetric_t> y_series;
+    ComplexTensor<asymmetric_t> y_shunt;
+
+    ComplexTensor<asymmetric_t> kron_reduction(const ComplexTensor4& matrix_to_reduce) const {
+        ComplexTensor4 Y = matrix_to_reduce;
+        ComplexTensor<asymmetric_t> Y_aa = ComplexTensor<asymmetric_t>(Y(0,0), Y(1,0), Y(1, 1), Y(2,0), Y(2,1), Y(2,2));
+        ComplexValue<asymmetric_t> Y_ab(Y(0,3), Y(1,3), Y(2,3));
+        Eigen::Array<DoubleComplex, 1, 3> Y_ba;
+        Y_ba << Y(3,0), Y(3,1), Y(3,2);
+        DoubleComplex Y_bb_inv = 1.0 / Y(3,3);
+
+        return Y_aa - ((Y_ab * Y_bb_inv).matrix() * Y_ba.matrix()).array();
+    }
+
+    ComplexTensor<asymmetric_t> compute_z_series_from_input(const power_grid_model::AsymLineInput& asym_line_input) {
+        ComplexTensor<asymmetric_t> z_series_abc;
+        if (is_nan(asym_line_input.r_na) && is_nan(asym_line_input.x_na)) {
+            ComplexTensor<asymmetric_t> r_matrix = ComplexTensor<asymmetric_t>(asym_line_input.r_aa, asym_line_input.r_ba, asym_line_input.r_bb, asym_line_input.r_ca, asym_line_input.r_cb, asym_line_input.r_cc);
+            ComplexTensor<asymmetric_t> x_matrix = ComplexTensor<asymmetric_t>(asym_line_input.x_aa, asym_line_input.x_ba, asym_line_input.x_bb, asym_line_input.x_ca, asym_line_input.x_cb, asym_line_input.x_cc);
+            z_series_abc = r_matrix + x_matrix * 1.0i;
+        } 
+        else {
```

Comment:
formatting can be automatically solved using `pre-commit` (see https://power-grid-model.readthedocs.io/en/stable/contribution/CONTRIBUTING.html#pre-commit-hooks ; you can apply the format changes retroactively using `pre-commit run --all-files` and run it before committing using `pre-commit install`)

```suggestion
        } else {
```

Reply:
I will try this once I've completly finished the implementation